scripts/natural_language_inference/main_intra.py

```diff
@@ -0,0 +1,217 @@
+        return pred
+
+def parse_args():
+    """
+    Parsing arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--data_root',
+                        help='root of data file', default='./data')
+    parser.add_argument('--train_set',
+                        help='training set file', default='snli_1.0/snli_1.0_train.txt')
+    parser.add_argument('--dev_set',
+                        help='development set file', default='snli_1.0/snli_1.0_dev.txt')
+    parser.add_argument('--batch_size',
+                        help='batch size', default=32, type=int)
+    parser.add_argument('--print_period',
+                        help='the interval of two print', default=20, type=int)
+    parser.add_argument('--checkpoints',
+                        help='path to save checkpoints', default='checkpoints')
+    parser.add_argument('--model',
+                        help='model file to test, only for test mode', default=None)
+    parser.add_argument('--mode',
+                        help='train or test', default='train')
+    parser.add_argument('--lr',
+                        help='learning rate', default=0.025, type=float)
+    parser.add_argument('--weight_decay',
+                        help='weight decay', default=0.0001, type=float)
+    parser.add_argument('--maximum_iter',
+                        help='maximum number of iterations to train',
+                        default=200, type=float)
+    parser.add_argument('--embedding',
+                        help='word embedding type',
+                        default='glove')
+    parser.add_argument('--embedding_source',
+                        help='embedding file soure',
+                        default='glove.6B.300d')
+    parser.add_argument('--embedding_size',
+                        help='size of embedding. Change it when using new embedding file!',
+                        default=300, type=int)
+
+    return parser.parse_args()
+
+def train_network(model, train_set, embedding, ctx, args):
+    """
+    Train the network.
+    """
+    model.collect_params().initialize(
+        mx.init.Xavier(), force_reinit=True, ctx=ctx)
+    celoss = gluon.loss.SoftmaxCrossEntropyLoss()
+    trainer = gluon.Trainer(model.collect_params(), 'adagrad',
+                            {'learning_rate': args.lr, 'wd': args.weight_decay})
+    batch_size = args.batch_size
+    print_period = args.print_period
+    if not os.path.exists(args.checkpoints):
+        os.mkdir(args.checkpoints)
+    emb_layer = mx.gluon.nn.Embedding(len(embedding.idx_to_token),
+                                      embedding.idx_to_vec[0].size)
+    emb_layer.initialize()
+    emb_layer.weight.set_data(embedding.idx_to_vec)
+    padder = nlp.data.batchify.Pad()
+    model.save_parameters(
+        os.path.join(args.checkpoints, 'epoch-0.gluonmodel'))
+    for epoch in range(1, args.maximum_iter + 1):
+        access_key = list(range(len(train_set)))
```

Comment:
I think shallow copying `train_set` and shuffling it is easier to read and as efficient than this. If we are staying with this, the variable name `access_key` is a bit unconventional. I would suggest something like `data_indices` so that it is more clear what this list is about.